src/main/java/ru/ayaz/UserMessageHandler.java

```diff
@@ -0,0 +1,142 @@
+    private MessageParser parser;
+
+    public UserMessageHandler(int queueSize) {
+        usersMap = new HashMap<String, User>();
+        outputWriterMap = new HashMap<String, PrintWriter>();
+        inputReaderMap = new HashMap<String, BufferedReader>();
+        messageQueue = new ArrayBlockingQueue(queueSize, true);
+        parser = new MessageParser();
+    }
+
+    public void enqueueMessage(UserMessage message) {
+        try {
+            messageQueue.put(message);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public boolean queueContainsMessage(UserMessage message) {
+        return messageQueue.contains(message);
+    }
+
+    public UserMessage takeMessage() throws InterruptedException {
+        return messageQueue.take();
+    }
+
+    public void addUserToMap(User user) {
+        usersMap.put(user.getNickname(), user);
+    }
+
+    public void addWriterToMap(String nickname, PrintWriter writer) {
+        outputWriterMap.put(nickname, writer);
+    }
+
+    public void addReaderToMap(String nickname, BufferedReader reader) {
+        inputReaderMap.put(nickname, reader);
+    }
+
+
+    @Override
+    public void run() {
+        try {
+            while (true) {
+                UserMessage message = takeMessage();
+                if (parser.isCommand(message.getText())) {
+                    executeCommand(message);
+                } else {
+                    sendMessageToEveryone(message);
+                }
+            }
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void executeCommand(UserMessage commandMessage) {
+        String nickname = commandMessage.getSenderName();
+        String text = commandMessage.getText();
+
+        switch (parser.getCommand(text)) {
+            case "#quit":
+                executeQuitCommand(commandMessage);
+                break;
+            case "#ignore":
+                executeIgnoreCommand(commandMessage);
+                break;
+            default:
+                PrintWriter writer = outputWriterMap.get(nickname);
+                sendMessageToExactPerson("Unknown command", writer);
+                break;
+        }
+    }
+
+
+    private void executeQuitCommand(UserMessage commandMessage) {
+        closeWriterAndReaderInMap(commandMessage.getSenderName());
+    }
+
+
+    private void executeIgnoreCommand(UserMessage commandMessage) {
+        String nickname = commandMessage.getSenderName();
+        String text = commandMessage.getText();
+
+        User user = usersMap.get(nickname);
+        try {
+            user.ignoreUser(parser.getFirstArgument(text));
+        } catch (InvalidUserCommandException e) {
+            PrintWriter writer = outputWriterMap.get(nickname);
+            sendMessageToExactPerson("Invalid argument", writer);
+        }
+    }
+
+
+    private void closeWriterAndReaderInMap(String nickname) {
```

Comment:
&#x1F34E; **Issue**
This method could be named better. Just from looking at the name question pops up: "What is this Map ?"